piptools/sync.py

```diff
@@ -59,7 +60,7 @@ def dependency_tree(
 
     while queue:
         v = queue.popleft()
-        key = v.key
+        key = str(canonicalize_name(v.key))
```

Comment:
AFK: I'd prefer explicit `NormalizedName` type instead of str conversion, so I'll refactor this in a following up PR.